### ISSUE TITLE
[Java] Type constraints aren't carried over to java

### ISSUE THREAD

**Issue by TheDrawingCoder-Gamer** [HaxeFoundation/haxe#10331]
For generic interfaces and classes, exporting them to java doesn't seem to work. 
Example project: https://github.com/TheDrawingCoder-Gamer/java-jank/tree/master
In short, what happens is a java interface IBar has a type constraint, and when a haxe also defines a type constraint, it doesn't carry over, causing java to throw an compilation error
In java: `interface IBar<T extends Foo>`
In haxe: `class Example<T:Foo> implements IBar<T>`
java error: 
```
src\haxe\root\Example.java:6: error: type argument T#1 is not within bounds of type-variable T#2
public class Example<T> extends haxe.lang.HxObject implements java.jank.IBar<T>
```

This behavior is not seen with JVM, as it skips java and directly compiles to JVM

This shouldn't prevent this being fixed as Java is better for some things like Gradle.

**Comment by Simn** [comment]
The Java target will be put in "regression fixes only" state with the Haxe 4.3.0 release. See https://github.com/HaxeFoundation/haxe/wiki/Target-Support-Level#java-and-c-since-430 for additional information. We apologize for any inconvenience this causes!

**Comment by EliteMasterEric** [comment]
As much as I would prefer to use the `jvm` target, until #11016 is closed and the `jvm` target can generate valid bytecode for newer Java versions, I must use the `java` target for my application. As such, this issue is a fairly annoying blocker for me.
